src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,181 @@
+        command = in.nextLine();
+
+        while (!command.toLowerCase().equals("bye")) {
+            if (command.toLowerCase().startsWith("todo")) {
+                computeTodoCommand(tasks, command);
+            } else if (command.toLowerCase().startsWith("deadline")) {
+                computeDeadlineCommand(tasks, command);
+            } else if (command.toLowerCase().startsWith("event")){
+                computeEventCommand(tasks, command);
+            } else if (command.toLowerCase().equals("list")) {
+                printList(tasks);
+            } else if (command.toLowerCase().startsWith("done")) {
+                markAsDone(tasks, command);
+            } else if (command.toLowerCase().startsWith("delete")) {
+                removeTask(tasks, command);
+            } else {
+                printErrorMessage("Invalid command given. Please enter a new command.");
+            }
+            FileManager.writeToFile(tasks);
+            command = in.nextLine();
+        }
+        showFarewellMessage();
+    }
+
+    private static void markAsDone(ArrayList<Task> tasks, String command) {
+        int taskNum = -1;
+        boolean proceedOnwards = false;
+        try {
+            taskNum = Integer.parseInt(command.substring(5));
+            proceedOnwards = true;
+        } catch (StringIndexOutOfBoundsException e) {
+            printErrorMessage("Please provide a valid number.");
+        }
+
+        if (proceedOnwards) {
+            try {
+                tasks.get(taskNum - 1).setDone();
+                System.out.println(Duke.horizLine);
+                System.out.println("Nice! I've marked this task as done:\n" + "[\u2713] " + tasks.get(taskNum - 1).getDescription());
+                System.out.println(Duke.horizLine);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printErrorMessage("Number given is not in the list.");
+            } catch (IndexOutOfBoundsException e) {
+                printErrorMessage("Please provide a valid number.");
+            }
+        }
+    }
+
+    private static void printList(ArrayList<Task> tasks) {
+        System.out.println(Duke.horizLine);
+        System.out.println("Here are the tasks on your list: ");
+        for (int i = 0; i < tasks.size() ; i++) {
+            System.out.println((i+1) + ". " + tasks.get(i).toString());
+        }
+        System.out.println(Duke.horizLine);
+    }
+
+    private static void computeEventCommand(ArrayList<Task> tasks, String command) {
+        String[] eventDetails = new String[2];
+        try {
+            extractCommandDetails(eventDetails, command, "Event");
+            printAcknowledgement();
+            tasks.add(new Event(eventDetails[0], eventDetails[1]));
+            System.out.println(tasks.get(tasks.size() - 1).toString());
+            System.out.println("There are now " + tasks.size() + " tasks in the list.");
+            System.out.println(Duke.horizLine);
```

Comment:
Perhaps the print statements can be extracted out into a method to improve code readability :) Noticed that you are using the 3 same `println `repetitively.